### ISSUE TITLE
decoder_inputs' first element is not matched with GO flag's index

### ISSUE THREAD

**Issue by Harmonicahappy** [emedvedev/attention-ocr#131]
First of all, you define `GO` flag's index as 1 in `data_gen.py`:
```python
class DataGen(object):
    GO_ID = 1
    EOS_ID = 2
    IMAGE_HEIGHT = 32
    CHARMAP = ['', '', ''] + list('0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ')
```
Second,  `decoder_inputs` will be filled with value 0 in `model.py`:
```python
self.decoder_inputs = []
self.target_weights = []
for i in xrange(self.decoder_size + 1):
    self.decoder_inputs.append(
        tf.tile([0], [num_images])
    )
```
In train mode, it is ok because `decoder_inputs` will be filled with ground truth labels, but in predict mode, first element of `decoder_inputs` will be 0, and it is not the correct index for `GO` flag. Am I right？

**Comment by emedvedev** [comment]
Fixed in https://github.com/emedvedev/attention-ocr/pull/132

**Comment by kspook** [comment]
In my case, I didn't use CHARMAP.   I omit ['','',''] in the CHARMAP.
So, new code provides low accuracy. But old code provides good accuracy

1. result of new code
(py36) D:\attention-ocr_b36uwi>python aocr36 test dataset/testingwk.tfrecords
2019-04-17 16:28:37.505456: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
 2019-04-17 16:28:39,871 root  INFO     Step 1 (0.186s). Accuracy:  0.00%, loss: 25.585348, perplexity: 1.29293e+11, probability: 100.00%   0% (신한 vs ) dataset/word-test/kookmin3.jpg
2019-04-17 16:28:39,899 root  INFO     Step 2 (0.027s). Accuracy:  0.00%, loss: 25.671883, perplexity: 1.4098e+11, probability: 100.00%   0% (신한 vs ) dataset/word-test/kookmin.jpg
2019-04-17 16:28:39,920 root  INFO     Step 3 (0.020s). Accuracy:  0.00%, loss: 0.000002, perplexity: 1.00000, probability: 99.99%   0% (신한 vs 신한) dataset/word-test/shinhan3.jpg
2019-04-17 16:28:39,941 root  INFO     Step 4 (0.020s). Accuracy:  0.00%, loss: 25.836084, perplexity: 1.66138e+11, probability: 100.00%   0% (신한 vs ) dataset/word-test/woori.jpg
2019-04-17 16:28:39,962 root  INFO     Step 5 (0.019s). Accuracy:  0.00%, loss: 25.663200, perplexity: 1.39761e+11, probability: 100.00%   0% (신한 vs ) dataset/word-test/kookmin2.jpg
2019-04-17 16:28:39,983 root  INFO     Step 6 (0.019s). Accuracy:  0.00%, loss: 0.000002, perplexity: 1.00000, probability: 100.00%   0% (신한 vs 신한) dataset/word-test/shinhan2.jpg
2019-04-17 16:28:40,005 root  INFO     Step 7 (0.020s). Accuracy:  0.00%, loss: 0.000002, perplexity: 1.00000, probability: 100.00%   0% (신한 vs 신한) dataset/word-test/shinhan.jpg


2. result of old code
2019-04-18 10:12:37.148256: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
2019-04-18 10:12:39,491 root  INFO     Step 1 (0.194s). Accuracy: 100.00%, loss: 0.000002, perplexity: 1.00000, probability: 99.99% 100% (신한)
2019-04-18 10:12:39,512 root  INFO     Step 2 (0.020s). Accuracy: 100.00%, loss: 0.000003, perplexity: 1.00000, probability: 99.99% 100% (신한)
2019-04-18 10:12:39,534 root  INFO     Step 3 (0.019s). Accuracy: 100.00%, loss: 0.000002, perplexity: 1.00000, probability: 99.99% 100% (신한)
2019-04-18 10:12:39,554 root  INFO     Step 4 (0.019s). Accuracy: 100.00%, loss: 0.000003, perplexity: 1.00000, probability: 99.99% 100% (신한)
2019-04-18 10:12:39,575 root  INFO     Step 5 (0.020s). Accuracy: 100.00%, loss: 0.000003, perplexity: 1.00000, probability: 99.99% 100% (신한)
2019-04-18 10:12:39,596 root  INFO     Step 6 (0.020s). Accuracy: 100.00%, loss: 0.000003, perplexity: 1.00000, probability: 99.99% 100% (신한)
2019-04-18 10:12:39,619 root  INFO     Step 7 (0.021s). Accuracy: 100.00%, loss: 0.000002, perplexity: 1.00000, probability: 99.99% 100% (신한)
2019-04-18 10:12:39,640 root  INFO     Step 8 (0.020s). Accuracy: 100.00%, loss: 0.000002, perplexity: 1.00000, probability: 99.99% 100% (신한)
2019-04-18 10:12:39,661 root  INFO     Step 9 (0.020s). Accuracy: 100.00%, loss: 0.000002, perplexity: 1.00000, probability: 99.99% 100% (신한)
2019-04-18 10:12:39,682 root  INFO     Step 10 (0.020s). Accuracy: 100.00%, loss: 0.000002, perplexity: 1.00000, probability: 99.99% 100% (신한)
